Mathlib/CategoryTheory/GuitartExact/VerticalComposition.lean

```diff
@@ -119,6 +120,46 @@ instance vComp' [GuitartExact w] [GuitartExact w'] {L₁₂ : C₁ ⥤ C₃}
   dsimp only [TwoSquare.vComp']
   infer_instance
 
+lemma vComp_iff_of_equivalences (eL : C₂ ≌ C₃) (eR : D₂ ≌ D₃)
+    (w' : H₂ ⋙ eR.functor ≅ eL.functor ⋙ H₃) :
+    (w.vComp w'.hom).GuitartExact ↔ w.GuitartExact := by
+  constructor
+  · intro hww'
+    letI : CatCommSq H₂ eL.functor eR.functor H₃ := ⟨w'⟩
+    have hw' : CatCommSq.iso H₂ eL.functor eR.functor H₃ = w' := rfl
+    letI : CatCommSq H₃ eL.inverse eR.inverse H₂ := CatCommSq.vInvEquiv _ _ _ _ inferInstance
+    let w'' := CatCommSq.iso H₃ eL.inverse eR.inverse H₂
+    let α : (L₁ ⋙ eL.functor) ⋙ eL.inverse ≅ L₁ :=
+      Functor.associator _ _ _ ≪≫ isoWhiskerLeft L₁ eL.unitIso.symm ≪≫ L₁.rightUnitor
+    let β : (R₁ ⋙ eR.functor) ⋙ eR.inverse ≅ R₁ :=
+      Functor.associator _ _ _ ≪≫ isoWhiskerLeft R₁ eR.unitIso.symm ≪≫ R₁.rightUnitor
+    have : w = (w.vComp w'.hom).vComp' w''.hom α β := NatTrans.ext _ _ (by
+      ext X₁
+      dsimp
+      simp? [w'', β, α] says
+        simp only [vComp'_app, Functor.comp_obj, Iso.trans_inv, isoWhiskerLeft_inv, Iso.symm_inv,
+          assoc, NatTrans.comp_app, Functor.id_obj, Functor.rightUnitor_inv_app, whiskerLeft_app,
+          Functor.associator_inv_app, comp_id, id_comp, vComp_app, Functor.map_comp,
+          Equivalence.inv_fun_map, Iso.trans_hom, isoWhiskerLeft_hom, Iso.symm_hom,
+          Functor.associator_hom_app, Functor.rightUnitor_hom_app, Iso.hom_inv_id_app_assoc,
+          w'', α, β]
+      erw [CatCommSq.vInv_iso'_hom_app, hw']
+      simp only [assoc, ← eR.inverse.map_comp_assoc]
+      erw [Equivalence.counitInv_app_functor, ← NatTrans.naturality_assoc, Iso.hom_inv_id_app,
+        comp_id, ← NatTrans.naturality_assoc, Iso.hom_inv_id_app_assoc, ← H₂.map_comp,
+        Iso.hom_inv_id_app, H₂.map_id, comp_id])
```

Comment:
I have managed to reduced this to only one `erw`:  such `erw [← NatTrans.naturality_assoc]` are basically impossible to avoid when the natural transformation involves compositions of functors (or the identity functor).